easybuild/easyblocks/n/namd.py

```diff
@@ -74,33 +85,42 @@ def configure_step(self):
         namd_comp = namd_comps.get(comp_fam, None)
         if charm_arch_comp is None or namd_comp is None:
             raise EasyBuildError("Unknown compiler family, can't complete Charm++/NAMD target architecture.")
+
+        # NOTE: important to add smp BEFORE the compiler
+        # charm arch style is: mpi-linux-x86_64-smp-mpicxx
+        # otherwise the setting of name_charm_arch below will get things
+        # in the wrong order
+        if self.toolchain.options.get('openmp', False):
+            self.cfg.update('charm_arch', 'smp')
         self.cfg.update('charm_arch', charm_arch_comp)
 
         self.log.info("Updated 'charm_arch': %s" % self.cfg['charm_arch'])
         self.namd_arch = '%s-%s' % (self.cfg['namd_basearch'], namd_comp)
         self.log.info("Completed NAMD target architecture: %s" % self.namd_arch)
 
-        charm_tarballs = glob.glob('charm-*.tar')
-        if len(charm_tarballs) != 1:
-            raise EasyBuildError("Expected to find exactly one tarball for Charm++, found: %s", charm_tarballs)
-
-        extract_file(charm_tarballs[0], os.getcwd())
 
         tup = (self.cfg['charm_arch'], self.cfg['charm_opts'], self.cfg['parallel'], os.environ['CXXFLAGS'])
-        cmd = "./build charm++ %s %s -j%s %s -DMPICH_IGNORE_CXX_SEEK" % tup
-        charm_subdir = '.'.join(os.path.basename(charm_tarballs[0]).split('.')[:-1])
+        cmd = "./build charm++ %s %s --with-numa -j%s %s -DMPICH_IGNORE_CXX_SEEK" % tup
+        charm_subdir = '.'.join(os.path.basename(self.charm_tarballs[0]).split('.')[:-1])
         self.log.debug("Building Charm++ using cmd '%s' in '%s'" % (cmd, charm_subdir))
         run_cmd(cmd, path=charm_subdir)
 
         # compiler (options)
         self.cfg.update('namd_cfg_opts', '--cc "%s" --cc-opts "%s"' % (os.environ['CC'], os.environ['CFLAGS']))
-        self.cfg.update('namd_cfg_opts', '--cxx "%s" --cxx-opts "%s"' % (os.environ['CXX'], os.environ['CXXFLAGS']))
+        self.cfg.update('namd_cfg_opts', '--cxx "%s --std=c++11" --cxx-opts "%s"' % (os.environ['CXX'], os.environ['CXXFLAGS']))
```

Comment:
tested, using --std=c++11 breaks older NAMD easyconfig, so let's inject this conditionally

done in https://github.com/akesandgren/easybuild-easyblocks/pull/5